dandiapi/api/copy.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+import math
+
+import boto3
+
+try:
+    from storages.backends.s3boto3 import S3Boto3Storage
+except ImportError:
+    # This should only be used for type interrogation, never instantiation
+    S3Boto3Storage = type('FakeS3Boto3Storage', (), {})
+try:
+    from minio_storage.storage import MinioStorage
+except ImportError:
+    # This should only be used for type interrogation, never instantiation
+    MinioStorage = type('FakeMinioStorage', (), {})
+
+from dandischema.digests.dandietag import PartGenerator, gb, tb
+
+
+@dataclass
+class CopyObjectResponse:
+    key: str
+    etag: str
+
+
+@dataclass
+class CopyPartGenerator(PartGenerator):
+    """Override the normal PartGenerator class to account for larger copy part sizes."""
+
+    @classmethod
+    def for_file_size(cls, file_size: int) -> CopyPartGenerator:
+        """Calculate sequential part sizes given a file size."""
+        if file_size == 0:
+            return cls(0, 0, 0)
+
+        part_size = gb(5)
```

Comment:
5GB is the max part size for s3 multipart upload/copy. Since this is an S3 bucket to bucket copy, and nothing passes through the client, this seems the most efficient. 64mb is probably fine when uploading from a client to S3, but seems wildly inefficient at the speeds that S3 is capable of.

Reply:
For anybody else, after some offline discussion, we've decided to keep the current 64mb part size for now. If that causes serious bandwidth issues, we can re-evaluate.